kotlin/integrated-windows-authentication/src/main/java/com/esri/arcgisruntime/sample/integratedwindowsauthentication/MainActivity.kt

```diff
@@ -0,0 +1,313 @@
+        mapView.map = ArcGISMap(Basemap.createStreets())
+
+        // Set authentication challenge handler
+        AuthenticationManager.setAuthenticationChallengeHandler(this)
+
+        // Set up recycler view for listing portal items
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        PortalItemAdapter(this).let {
+            this.portalItemAdapter = it
+            recyclerView.adapter = it
+        }
+
+        searchPublicButton.setOnClickListener {
+            // Search the the public ArcGIS portal
+            searchPortal(Portal(getString(R.string.arcgis_url)))
+        }
+
+        searchSecureButton.setOnClickListener {
+            // Get the string entered for the secure portal URL.
+            portalUrlEditText.text?.toString()?.let {
+                // If the entered URL is a valid URL
+                if (Patterns.WEB_URL.matcher(it).matches()) {
+                    searchPortal(Portal(portalUrlEditText.text.toString(), true))
+                } else {
+                    getString(R.string.error_portal_url).let { errorString ->
+                        Toast.makeText(this, errorString, Toast.LENGTH_LONG).show()
+                        Log.e(TAG, errorString)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun searchPortal(portal: Portal) {
+        // Hide portal list during search
+        recyclerView.visibility = View.INVISIBLE
+
+        // Show portal load state during search
+        portalLoadStateView.visibility = View.VISIBLE
+        portalLoadStateTextView.text = getString(R.string.portal_load_state_searching, portal.uri)
+
+        // Add Runnable to execute when Portal has finished loading
+        portal.addDoneLoadingListener {
+            if (portal.loadStatus == LoadStatus.LOADED) {
+                // Update load state in UI with the portal URI
+                portalLoadStateTextView.text = getString(R.string.portal_load_state_connected, URI(portal.uri).host)
+
+                // Report the user name used for this connection.
+                portal.user?.let {
+                    portalLoadStateTextView.text = getString(
+                        R.string.portal_user_connected,
+                        if (it.username != null) it.username else getString(R.string.portal_user_anonymous)
+                    )
+                }
+
+                // Search the portal for web maps
+                portal.findItemsAsync(PortalQueryParameters("type:(\"web map\" NOT \"web mapping application\")"))
+                    ?.let { portalItemResult ->
+                        portalItemResult.addDoneListener {
+                            try {
+                                portalItemResult.get()?.results?.let { portalItemSetResults ->
+                                    portalItemAdapter.updatePortalItems(portalItemSetResults)
+                                }
+                            } catch (executionException: ExecutionException) {
```

Comment:
Can you catch an ExecutionException || Interrupted exception (Or even just an Exception) and return message from them in a single catch?

Reply:
Not in Kotlin I'm afraid: https://youtrack.jetbrains.com/issue/KT-7128